### PR TITLE
Invariant Metrics

### PR DESCRIPTION
**Pull request by nguigs**
Compute Riemann curvature on lie groups with an invariant metric:
- renamed inner_product_matrix to metric_matrix, because it's shorter
- Add invariant metrics represented by a diagonal matrix (#702 )
This is currently tested on SO(n), but will be tested on SE(n) in a next PR: need to create its Lie Algebra first

### CODE REVIEW

**Comment by nguigs** [comment]
I thought we removed this DeepSource issue @ninamiolane ?

File: geomstats/geometry/beta_distributions.py [hunk 1]
```diff
@@ -207,7 +207,7 @@ def metric_det(param_a, param_b):
                                                   gs.polygamma(1, param_b))
         return metric_det
 
-    def inner_product_matrix(self, base_point):
+    def metric_matrix(self, base_point):
```
Comment (ninamiolane) [inline geomstats/geometry/beta_distributions.py:210]: Good idea!

File: geomstats/geometry/invariant_metric.py [hunk 2]
```diff
@@ -163,11 +166,352 @@ def inner_product_matrix(self, base_point=None):
 
         metric_mat = gs.einsum(
             '...ij,...jk->...ik',
-            inv_jacobian_transposed, self.inner_product_mat_at_identity)
+            inv_jacobian_transposed, self.metric_mat_at_identity)
         metric_mat = gs.einsum(
             '...ij,...jk->...ik', metric_mat, inv_jacobian)
         return metric_mat
 
+    def structure_constant(self, tangent_vec_a, tangent_vec_b, tangent_vec_c):
```
Comment (ninamiolane) [inline geomstats/geometry/invariant_metric.py:174]: Nice!

File: geomstats/geometry/invariant_metric.py [hunk 3]
```diff
@@ -163,11 +166,352 @@ def inner_product_matrix(self, base_point=None):
...
+            Tangent vector at identity.
+
+        Returns
+        -------
+        structure_constant : array-like, shape=[...,]
+        """
+        bracket = GeneralLinear.bracket(tangent_vec_a, tangent_vec_b)
+        return self.inner_product_at_identity(bracket, tangent_vec_c)
+
+    def adjoint_star(self, tangent_vec_a, tangent_vec_b):
```
Comment (ninamiolane) [inline geomstats/geometry/invariant_metric.py:196]: NIT: why not calling it `dual_adjoint`, thus refering to its name and not its notation?

**Review by ninamiolane — APPROVED** [review]
Nice! some minor comments in the review.

Some questions:
- Why is coverage decreased?
- I remember from discussions with Xavier that non-diagonal invariant metric geodesic were not easy to compute, and the strategy that I had in mind for having them in geomstats was not appropriate. Did you check with him on this?

**Comment by ninamiolane** [comment]
Yes, let's forget about this deepsource error, I don't know why it is still showing up...

**Comment by nguigs** [comment]
>     * Why is coverage decreased?

I think some things are not tested with Tensorflow, I will investigate!

>     * I remember from discussions with Xavier that non-diagonal invariant metric geodesic were not easy to compute, and the strategy that I had in mind for having them in geomstats was not appropriate. Did you check with him on this?

Yes, they are not straightforward, we will do it with integration of Euler Poincarré equations, as [in my repo](https://github.com/nguigs/ladder-methods/blob/master/sen_tools/sen.py).

File: geomstats/geometry/invariant_metric.py (hunk 3 above)
Comment (nguigs) [reply geomstats/geometry/invariant_metric.py:196]: yes much better

**Pull request merged**